File: control-flow/Q89/main.go
package main

import "fmt"

func main() {
	var num int
	fmt.Print("Enter a number: ")
	fmt.Scan(&num)
	pow := 10
	temp := num
	reversedNum := 0
	for temp > 0 {
		reversedNum = reversedNum*pow + temp%10
		temp /= 10
	}
	if reversedNum == num {
		fmt.Println("YES")
	} else {
		fmt.Println("NO")
	}
}

File: control-flow/Q89/main.rs
use std::io;

fn main() {
    let mut input = String::new();

    println!("Enter a number:");
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let num: i32 = input.trim().parse().expect("Not a valid number");

    let pow: i32 = 10;
    let mut temp = num;
    let mut reversed_num: i32 = 0;

    while temp > 0 {
        reversed_num = reversed_num * pow + (temp % 10);
        temp /= 10;
    }

    if reversed_num == num {
        println!("YES");
    } else {
        println!("NO");
    }
}